boilerplate/bin/tools/patch.mjs

```diff
@@ -0,0 +1,26 @@
+/** @callback Replacement @param search {string} @returns {string} */
+/**
+ * Insert a string into a file based on a search string.
+ * @param file {string} contents of the file
+ * @param search {string} string to search for in the file
+ * @param replacement {Replacement} callback to construct to replacement string
+ * @returns {string} string with the replacement made
+ */
+export const insert = (file, search, replacement) => file.replace(search, replacement(search))
+
+/**
+ * @callback UpdateCallback
+ * @param file {string} contents of the file
+ * @returns {string} updated file contents
+ */
```

Comment:
`zx` can be used with typescript, but it seems to take a lot wrangling to set it up. However, the existing globals have nice type defs already. And we can use jsdoc to create types otherwise. Another approach here may be to add `.d.ts` files and import them using `jsdoc`